source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -65,6 +67,29 @@ Extractor::Extractor(const envoy::api::v2::filter::http::Extraction &extractor)
         fmt::format("group {} requested for regex with only {} sub groups",
                     group_, extract_regex_.mark_count()));
   }
+
+  switch (mode_) {
+    case ExtractionApi::EXTRACT:
+      extraction_func_ = std::bind(&Extractor::extractValue, this, _1, _2);
+      break;
+    case ExtractionApi::SINGLE_REPLACE:
+      if (!replacement_text_.has_value()) {
+        throw EnvoyException("SINGLE_REPLACE mode set but no replacement text provided");
+      }
+      extraction_func_ = std::bind(&Extractor::replaceIndividualValue, this, _1, _2);
+      break;
+    case ExtractionApi::REPLACE_ALL:
+      if (!replacement_text_.has_value()) {
+        throw EnvoyException("REPLACE_ALL mode set but no replacement text provided");
+      }
+      if (group_ != 0) {
+        throw EnvoyException("REPLACE_ALL mode set but subgroup is not 0");
+      }
+      extraction_func_ = std::bind(&Extractor::replaceAllValues, this, _1, _2);
+      break;
+    default:
+      throw EnvoyException("Unknown mode");
```

Comment:
this is probably fine, but you could also consider using `PANIC_DUE_TO_CORRUPT_ENUM` here